specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-04-01/managedClusters.json

```diff
@@ -1483,6 +1483,10 @@
           "type": "string",
           "description": "Version of node image"
         },
+        "upgradeSettings": {
+          "$ref": "#/definitions/AgentPoolUpgradeSettings",
+          "description": "Settings for upgrading the agentpool"
+        },
```

Comment:
This is a new field being added, and considered breaking change. This will break the customers using older version of the SDK

Reply:
Shouldnt' be  a breaking change since it's fine for this field to be empty/not included. Is there something I need to add to denote this?

Reply:
Here' another example wher we added a field but are totally fine with it being empty
https://github.com/Azure/azure-rest-api-specs/pull/8865

Reply:
Oh maybe you think this is an already released api because it's april. We haven't released the april api. (slipped due to covid) so this is only new in the new api.

Reply:
1. Will this break old SDK's for this API version? Do you've telemetry data to prove?
2. Can older SDK cause overwrite of this parameter value while doing GET/PUT? 

---
In reply to: [424026074](https://github.com/Azure/azure-rest-api-specs/pull/9331#discussion_r424026074) [](ancestors = 424026074)

Reply:
In older versions of the API this field is neither returned nor accepted.

Reply:
And by not accepted I mean it will be completely ignored

Reply:
As Paul mentioned this is a net new property and net new feature capability, so it can't conflict with older API versions.

We're under a bit of a deadline to get this merged, happy to get on a call if the above still isn't clear so we can merge today @chiragg4u